runbotics/runbotics-orchestrator-ui/src/src-app/views/variable/VariableListView/Table/Table.tsx

```diff
@@ -53,25 +72,83 @@ const Table: FunctionComponent<TableProps> = ({ className, setVariableDetailStat
         onEdit,
         hasDeleteVariableAccess,
         hasEditVariableAccess,
-        globalVariables,
+        globalVariables
     });
 
     useEffect(() => {
-        dispatch(globalVariableActions.getGlobalVariables());
-        // eslint-disable-next-line react-hooks/exhaustive-deps
+        const queryPage = parseInt(router.query.page as string) || 0;
+        const queryPageSize = parseInt(router.query.pageSize as string) || 10;
+        const querySearch = router.query.search as string || '';
+        const querySortField = router.query.sortField as string || 'lastModified';
+        const querySortDirection = router.query.sortDirection || 'DESC';
```

Comment:
maybe we can move all the logic of paging into some hook? eg. usePagination<SortFields>() which keeps the state of page, pageSize, search, sortModel etc.